source/extensions/quic_listeners/quiche/envoy_quic_server_stream.cc

```diff
@@ -247,6 +247,13 @@ void EnvoyQuicServerStream::maybeDecodeTrailers() {
   }
 }
 
+bool EnvoyQuicServerStream::OnStopSending(quic::QuicRstStreamErrorCode error) {
```

Comment:
Can you also add same callback to EnvoyQuicClientStream?

Reply:
Per discussion on quic-dev I don't think we should.
If the server is to stop sending a response to the client, there's not much to do but reset.
If the client is to stop sending a request to the server, the server may be processing an error response which will come in, so I think resetting is not the correct action.

Reply:
Makes sense. But in IETF quic STOP_SENDING and RST_STREAM only close one side respectively. The reason we need to callback here is because read side is guarantee'ed to be closed before a server stream starts sending. Can you `ASSERT(read_side_closed())` or add 
```
if (read_side_closed()) {
  runResetCallbacks(quicRstErrorToEnvoyRemoteResetReason(error));
}
```